### ISSUE TITLE
Inaccurate focus of jumping table of contents and footnotes when using a screen reader

### ISSUE THREAD

**Issue by cary-rowen** [edrlab/thorium-reader#1379]
### description:
When reading a book with a table of contents and footnotes using Thorium-1.6 I found that I couldn't jump to where I should be, which worked fine in Thorium-1.5.

### Steps to reproduce:
1. Open the book and click on the "table of contents" button in the navigation, then click on "第四章手阳明经络与腧穴" ➡ "第一节 手阳明经络" ➡ "三、手阳明经别".
2. Click on the footnote link [1] in the text.

### Actual performance:
1. Perform step (1): The focus of the screen reader is at the beginning of the "main content".
2. Perform step (2): The screen reader focus is after the last footnote link [25] and does not read anything aloud.

### Expected performance:
1. Perform step (1): The focus of the screen reader is on "三、手阳明经别" in the text, and you can use the arrow keys to read the section.
2. Perform step (2): The focus of the screen reader is on the corresponding explanation part of the footnote link[1], and the explanation will be read out automatically.

### supplement:

- I am very sorry, the book used for testing is in Chinese, no English test sample was found, I don't know if I can upload the epub file of the book here, if I get your permission I will submit the test book here.
- Screen reader: NVDA2020.3.
- Windows version: Windows 10 20H2 (64-bit) build 19042.804
- Using the same version of screen reader, using the same test books, it behaves normally in Thorium-1.5
- Finally, I apologize for my poor English.

**Comment by danielweck** [comment]
Hello, thank you for reporting this bug.

I assume NVDA is already running before you start the Thorium application? 

I assume the footnote link is not a special EPUB popup footnote, but it is a normal HTML hyperlink which goes to the end of the current document? Or is this a link that targets another document in the publication?

I have a Windows computer with NVDA, so I could test the book myself. Can you share it privately? My email address is daniel.weck@gmail.com

PS: don't worry, your English is not poor :)

**Comment by cary-rowen** [comment]
Thank you for your patience in responding, NVDA was already running before starting Thorium, I am more than happy to share the book with you and would like to assist you in troubleshooting this issue. Again, thank you for your efforts.

**Comment by danielweck** [comment]
Thank you for sharing the EPUB. I tested with MacOS and Voice Over, the linking to the destination note at the end of the document (and back to the originating source hyperlink) is working. There is no loss of context.

I will try with Windows and NVDA.

**Comment by danielweck** [comment]
I ran some tests on Windows 10 with NVDA. I am not able to reproduce the loss of focus bug. The screen reader gets very confused by the Chinese text and skips content and sometimes even stops talking. However I can successfully link back and forth with the note at the end of the document, without loosing the reading position in the screen reader cursor.
I will also try with Thorium 1.5

**Comment by danielweck** [comment]
I have just finished testing with Thorium 1.5, and I am getting the same experience as with Thorium 1.6
Sorry, I am not able to reproduce the bug with NVDA

**Comment by cary-rowen** [comment]
Hello!

Thank you for your patience in testing, I have gone through my screen reader settings again and tried multiple scenarios as well as testing under narrator and was still able to reproduce the problem, but once or twice during my testing it worked as expected, but the vast majority did not work as expected. Can you connect to my computer via the NVDA remote add-on to test? Or instead, can I connect to your computer via the NVDA Remote Add-on to try to reproduce the problem?

Thanks.

**Comment by cary-rowen** [comment]
Can you use the same steps to test repeatedly? Or can I try to record a video to demonstrate this problem?
@danielweck

**Comment by danielweck** [comment]
Hello, although I have not personally been able to reproduce the bug consistently, I received feedback from other screen reader users who are experiencing mostly broken behaviour, and only occasionally working interactions (as one would normally expect). This lack of predictability makes it hard to pin-point exactly what it happening. So right now let's continue to collect feedback in order to document the problem(s).

**Comment by cary-rowen** [comment]
Hello, thanks for your efforts, I'll be the first to comment here if I find anything new, for now I'll downgrade to version 1.5 to use the desired features.

**Comment by cary-rowen** [comment]
Hi, danielweck! 
Is there the latest news about this issue?
Thanks.

@danielweck

**Comment by danielweck** [comment]
Hello, there is an ongoing discussion regarding similar issues with screen readers, but no concrete solution yet: https://github.com/edrlab/thorium-reader/issues/1438

**Comment by cary-rowen** [comment]
I am very happy for the release of version 1.7, but I have to stay at version 1.5.
How can we make the screen reader experience better?

**Comment by cary-rowen** [comment]
Hello,

Screen reader users rely heavily on this software, can we increase the priority of this issue?

@danielweck

**Comment by cary-rowen** [comment]
Is there any news about this issue?

**Comment by danielweck** [comment]
Hello, have you tried the latest official release?
https://github.com/edrlab/thorium-reader/releases/tag/v1.7.2

We would also really appreciate if you could test the latest automated build, from time to time:
https://github.com/edrlab/thorium-reader/releases
For example we have just published some screen reader accessibility fixes.

The focus jump is difficult to reproduce consistently. It happens often, but we are not sure why it sometimes work just fine.

**Comment by cary-rowen** [comment]
Hello daniel,

Sorry I missed the release of this version and did not test it for the first time, As you said, I tested 1.7.2 and it seems that this version of this issue has really improved.
Thank you very much for your outstanding efforts. I will continue to use and observe, and I will close this Issue when appropriate. Thank you again.

Cheers

**Comment by danielweck** [comment]
Thorium has evolved quite a bit since feedback was last reported (new Electron and Chromium versions, as well as fixed accessible labels on UI controls). I am closing this issue but feel free to re-open or to open a new one with a laser-focused bug report / behaviour description so that we can try to replicate at our end :)
Thank you very much!

**Comment by cary-rowen** [comment]
Thank you for your efforts, it should be closed.
